monasca_rsyslog/client.py

```diff
@@ -77,20 +77,25 @@ def __init__(self):
         self._min_poll_delay = cfg.CONF.api.min_poll_delay
         self._max_batch_size = cfg.CONF.api.max_batch_size
 
-    def _get_monasca_log_api_url(self, version='v3.0'):
+    def _get_monasca_log_api_url(self):
         """ Retrieves monasca log api endpoint. """
 
-        endpoint = cfg.CONF.api.url
-        if not endpoint:
+        url = cfg.CONF.api.url
+        if not url:
             catalog = self._sess.auth.get_auth_ref(self._sess).service_catalog
             endpoint = catalog.url_for(
                 service_type=cfg.CONF.auth.service_type,
                 interface=cfg.CONF.auth.endpoint_type,
                 region_name=cfg.CONF.auth.region_name,
             )
-        if not endpoint.endswith(version):
-            endpoint += '/' + version
-        return endpoint
+            for item in jsonutils.loads(self._sess.get(endpoint)).get('elements'):
```

Comment:
Can you use an endpoint filter to replace this block? Eg. https://docs.openstack.org/keystoneauth/latest/using-sessions.html#service-discovery

min version can be 3.0